vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -318,9 +318,34 @@ public enum SelectionMode {
          *
          * @see GridSingleSelectionModel
          */
-        SINGLE {
-            @Override
-            protected <T> GridSelectionModel<T> createModel(Grid<T> grid) {
+        SINGLE,
+
+        /**
+         * Multiselection mode that maps to built-in
+         * {@link SelectionModel.Multi}.
+         *
+         * @see GridMultiSelectionModel
+         */
+        MULTI,
+
+        /**
+         * Selection model that doesn't allow selection.
+         *
+         * @see GridNoneSelectionModel
+         */
+        NONE;
+
+        /**
+         * Creates the selection model to use with this enum.
+         *
+         * @param <T>
+         *            the type of items in the grid
+         * @param grid
+         *            the grid to create the selection model for
+         * @return the selection model
+         */
+        protected <T> GridSelectionModel<T> createModel(Grid<T> grid) {
+            if (SINGLE.equals(this)) {
```

Comment:
Any reason not to use a `switch` here?

Reply:
Only a personal preferences. Changed to enhanced `switch`